pkg/plugin/sdk/v1alpha2/plugin/interface.go

```diff
@@ -14,6 +14,7 @@ import (
 // Plugin determines how an aspect Plugin should be implemented.
 type Plugin interface {
 	BEPEventCallback(event *buildeventstream.BuildEvent) error
+	Setup(properties []byte) error
```

Comment:
What do the bytes represent? Is it weird having bytes in the public API?

Reply:
This is public. I have been thinking about this since yesterday, and I could be convinced to change the API here to take a `map[string]interface{}`. The disadvantage is that we limit the users from parsing the bytes straight into a typed struct.

Reply:
I do agree that its a little weird to have `[]byte` on a public API like this. However it is the most convenient form for that data to be in. Makes it super easy for the plugin to parse it into its expected type. If we pass it through as something else then most plugins are just going to need to unmarshal it again creating extra work on their end